java/vector/src/main/codegen/templates/AbstractFieldWriter.java

```diff
@@ -98,6 +98,12 @@ public ListWriter list() {
   }
 
   @Override
+  public ListWriter list(int size) {
+    fail("FixedSizeList");
+    return null;
+  }
```

Comment:
By the way, arbitrary data should still work fine and default to regular lists - you have to specifically set up the fixed size list if you want to use it. If you know you have a fixed size list, you can use that optimization, otherwise you can just use regular lists that will work the same with slightly more overhead.